File: src/main/java/io/github/bot/terminal/application/domains/db_connection/entity/PostgresDbConnection.java
package io.github.bot.terminal.application.domains.db_connection.entity;


import io.github.bot.terminal.application.domains.db_connection.repository.DbConnectionRepository;


public class PostgresDbConnection extends DbConnection<PostgresDbConnectionDetails> {

    private PostgresDbConnectionDetails details;
    private DbConnectionRepository repository;

    public PostgresDbConnection(PostgresDbConnectionDetails details,
                                DbConnectionRepository repository) {
        this.details = details;
        this.repository = repository;

    }

    public PostgresDbConnectionDetails getDetails() {
        return details;
    }

    @Override
    public String getUrl() {
        return "jdbc:postgresql://" + details.getHost() + ":" + details.getPort() + "/" + details.getDbName();
    }

    @Override
    public String getClassName() {
        return "org.postgresql.Driver";
    }

    @Override
    public String getCheckSelect() {
        return "SELECT 1";
    }

    @Override
    public void delete() {
        repository.deleteById(details.id);
    }
}

File: src/main/java/io/github/bot/terminal/application/domains/db_connection/entity/PostgresDbConnection.kt
package io.github.bot.terminal.application.domains.db_connection.entity

import io.github.bot.terminal.application.domains.db_connection.repository.DbConnectionRepository

class PostgresDbConnection(override val details: PostgresDbConnectionDetails,
                           private val repository: DbConnectionRepository) : DbConnection<PostgresDbConnectionDetails>() {
    override val id: String
        get() = details.id

    override val url: String
        get() = "jdbc:postgresql://${details.host}:${details.port}/${details.dbName}"

    override val className: String
        get() = "org.postgresql.Driver"

    override val checkSelect: String
        get() = "SELECT 1"

    fun delete() {
        repository.deleteById(details.id)
    }

}